setup.sh

```diff
@@ -2,48 +2,79 @@
 # Sets up environment variables and such, assuming requirements already installed.
 # Kevin Svetlitski, rev. Jonathan Shi
 
-[ ! -f 'package.json' ] && echo 'You are in the wrong directory!' && exit 1
-[ -z "$VIRTUAL_ENV" ] && echo 'You must activate your virtualenv first!' && exit 1
+{ [ ! -d 'csm_web' ] || [ ! -f 'package.json' ]; } && echo 'You are in the wrong directory!' 1>&2 && exit 1
+[ -z "$VIRTUAL_ENV" ] && echo 'You must activate your virtualenv first!' 1>&2 && exit 1
+
+target_version=$(sed 's/[^0-9.]//g' runtime.txt)
+installed_version=$(python3 --version | sed 's/[^0-9.]//g')
+if [ "$target_version" != "$installed_version" ]
+then
+   echo "You have python version $installed_version installed, but the expected version is $target_version.
```

Comment:
Echo this at the end instead so it doesn't get lost in the output of npm install etc.